tartufo/scanner.py

```diff
@@ -609,6 +609,11 @@ def chunks(self) -> Generator[types.Chunk, None, None]:
                 branches = [
                     x for x in unfiltered_branches if x.name == self.git_options.branch
                 ]
+
+                if len(branches) == 0:
+                    raise BranchNotException(
+                        f"Branch {self.git_options.branch} was not found."
+                    )
```

Comment:
Well, `BranchNotException` irks me because it feels like we're missing a word... the branch isn't what? Found? Valid?

Worse, I went and looked to see what tartufo was using for a convention, and they're pretty generic:

```
class TartufoException(Exception):
class ConfigException(TartufoException):
class ScanException(TartufoException):
class GitException(TartufoException):
class GitLocalException(GitException):
class GitRemoteException(GitException):
```

This makes me think we probably should be raising `GitLocalException` for a local repo or `GitRemoteException` for a remote repo, but I get that we probably don't want to have to screw around trying to pick the right one.

My suggestion here therefore would be to use `GitBranchException` (which should be a specialization of `GitException`).

The other option that came to mind would be some sort of `UserFootInMouthException` for when we are whining about GIGO directives, but there doesn't seem any call for it at this point so I'd stick with something simple.

Reply:
I did not want to mark it as a GitException as it wasn't an issue interacting with Git (`Raised if there is a problem interacting with git`). This is a logic issue unrelated to interacting with git. For example, if someone typed in `mybrach` and missed the `n`, this is a user issue where it would say `mybrach not found`.

Reply:
Regardless of all else, I agree that `BranchNotException` doesn't particularly make sense. As @rbailey-godaddy said, it sounds like it's missing a word. I would recommend renaming it to `BranchNotFoundException` would be very appropriate.

Reply:
lol. You're right. Not sure what I was thinking. I changed to BranchNotFoundException.